### ISSUE TITLE
Unable to join table to another join as sibling, rather than nested

### ISSUE THREAD

**Issue by JacobT14** [sequelize/sequelize#9809]
So I am trying to avoid having to just do the query with replacements, but basically I have a query that I want to end up with that looks like this:

```
select *
from "sharedGameData" sg
left outer join endgame eg1 on eg1.game_id = sg.gameid and eg1.playerid = '8E179150DC1414CD'
inner join endgame eg2 on eg2.game_id = eg1.game_id

ORDER BY sg."ts" DESC NULLS LAST
```
Basically I have a sharedGameData object that contains shared info. But, each game has 4 children rows(endgame table). I want to find all games based off a playerid(with or without a limit).

Using the query above it returns almost instantly.

In sequelize I have defined my models, but that hard part I am trying to figure out is how to create sibling join that can reference the first includes.

i.e.

```
includeObj = [
          {
            model: db.sequelize.models.endgame,
            as: 'endgames',
            required: false,
            attributes: {},
            where: { playerid: root.id }
          },
          {
            model: db.sequelize.models.endgame,
            required: true,
            as: 'eg2',
            where: { '$endgames.game_id': { $col: '$eg2.game_id' } }
          }
        ]
```
Obviously my syntax is a little off because the second include is not currently possible(I think?) The only way I can get it to return the right data is if I include the second include on the first include, but that does a join within a join(which I do not want) and causes the query to take 6 seconds instead of about a second, and I have to go and cleanse it as the first join returns an object that contains the second join results(I would like it to have just the sibling object, that being all the rows of that game. If the first join object is there that is okay, but not needed).

I've been trying lots of different variations of the includes, but when it comes down to it, I need it to be a sibling, rather than a nested include in order for the query to run as fast as possible.

For reference, here are my models currently:

```
  sequelize.models.endgame.belongsTo(
  sequelize.models.sharedGameData,
  { foreignKey: 'id' },
  { as: 'endgames' }
)
sequelize.models.endgame.hasMany(
  sequelize.models.endgame,
  {
    foreignKey: 'game_id',
    sourceKey: 'game_id'
  },
  { as: 'eg2 }
)
sequelize.models.sharedGameData.hasMany(
  sequelize.models.endgame,
  { foreignKey: 'game_id' },
  { as: 'endgames' }
)
```
Maybe I am overcomplicating this? Previously I was just getting all the games, then looping through and getting the individual rows in NodeJS, but if I am able to get the first query(or something like that), it provides a significant speed boost(scaling the higher the limit is)

(if it matters, this is all living below a GraphQL API, as part of a resolver)

To sum up, in sequelize I would like if possible to get All details of a Shared Game Object if a player is one of the detail rows. I imagine this would be a lot simpler if I stored the list of players on the Shared Game Object, so maybe that is the route I will go if this isn't easy to solve.

**Comment by JacobT14** [comment]
Figured it out, but I get an error when adding a limit.  Not sure if it is something I am doing wrong, but I posted it on SO for help on that. 
https://stackoverflow.com/questions/51882042/sequelize-join-table-on-its-self
